### DISCUSSION TITLE
RQL querying metadata

### DISCUSSION THREAD

**Discussion by james-allcode** [ravendb/ravendb#14742] [Q&A]
Hey everyone, 

So I'd like to do a query using RQL that pulls back and checks which documents have the 'HasRevision' flag in @metadata.@flags
Strangely though I can't seem to make it work. For example the following: 
```
from User as u
select u.'@metadata'.'@collection' as Collection
```
Works correctly pulling back a column of Id and Collection
However, 
```
from User as u
select u.'@metadata'.'@flags' as Flags
```
Brings back no flags even though they exist, similarly
```
from @all_docs as ad
where ad.'@metadata'.'@collection' IN ('Scopes')
```
Pulls back everything from the 'Scopes' collection no problem
However, 
```
from @all_docs as ad
where ad.'@metadata'.'@flags' IN ('HasRevisions')
```
pulls back no results. Am I doing something obviously stupid here?

**Comment by ayende** [comment]
`@flags` isn't actually exposed at the storage level, so it isn't accessible during indexing. 
We stitch the value when you read it from the server
